src/main/java/app/controller/GameEngine.java

```diff
@@ -137,7 +138,7 @@ private boolean legalMove(Vector start, Vector end)
     {
         for (Boundary bdy : map.getBoundaries())
         {
-            if(bdy.isCrossed(start, end))
+            if(bdy.isCrossed(start, end) && BoundaryType.isSolid(bdy.getBoundaryType()) == true)
```

Comment:
` BoundaryType.isSolid(bdy.getBoundaryType()) == true`

replace with 

`BoundaryType.isSolid(bdy.getBoundaryType())`

as isSolid() returns a boolean?

Reply:
ah yes, that's kinda stupid